### PR TITLE
F1 ddpg

### PR DESCRIPTION
**Pull request by pjfernandecabo**
Adding new deep deterministic policy gradient algorithm (DDPG) focused on F1 robotic agent .

Objective: the DDPG algorithm belongs to the family of algorithms that try to find the most optimal policy. It is designed with actor-critic architecture:
Actor - It proposes an action given a state.
Critic - It predicts if the action is good (positive value) or bad (negative value) given a state and an action.

Both the actor and the critic are implemented with deep neural networks, which allow us to work with discrete or continuous actions and a camera sensor as input to multidimensional state space.

Files modification/creation:

------------Main directory:
/main_rlstudio.py: new lines in the get_environmet() function to get the actions from the yaml file

-----------Config directory:
/config/**config.yaml**: is modified to introduce all the new DDPG algorithm hyperparameters and new continuous actions and sensor camera as a state input data.

-----------agents directory:
/agents/**init.py**: Modification of the TrainerFactory class to exec the new DDPG algorithm

/agents/f1/**utils.py** modified fourth line to point to new path: from rl_studio.agents.f1 import settings and also new functions added to support F1 training: class colors, print_messages, render_params, save_agent_physics, save_stats_episodes

new file /rl_studio/agents/f1/**train_ddpg.py**: the DDPG trainer

-----------Algorithms dir
/agorithms/**algorithms_type.py** file: new line for ddpg

new file /algorithms/**ddpg.py:** contains all the logic and classes of the DDPG algorithm

----------Envs directory:
new file /envs/gazebo/f1/models/**f1_env_ddpg.py**: manages the DDPG algorithm environment (used for camera sensor or any other sensor)

/envs/gazebo/f1/**env_type.py**: a new line is created for the DDPG algorithm

/envs/gazebo/f1/models/**init.py**: the new DDPG environment is added to the F1Env class

### CODE REVIEW

File: rl_studio/agents/__init__.py [hunk 1]
```diff
@@ -74,7 +83,9 @@ def __new__(cls, config):
             return CartpoleInferencer(config)
 
         elif agent == AgentsType.MOUNTAIN_CAR.value:
-            from rl_studio.agents.mountain_car.inference_qlearn import MountainCarInferencer
+            from rl_studio.agents.mountain_car.inference_qlearn import (
```
Comment (rubenlucas93) [inline rl_studio/agents/__init__.py:86]: Deberíamos elegir una manera u otra de importar no? Por mantener coherencia en el fichero.
Cual crees que es mejor?
import { fichero }
o 
import fichero
¿?

File: rl_studio/main_rlstudio.py [hunk 2]
```diff
@@ -18,7 +17,11 @@ def get_environment(config_file: dict, input_env: str) -> dict:
     return {
         "name": input_env,
         "params": config_file["environments"][input_env],
-        "actions": config_file["actions"]["available_actions"][input_env],
+        "actions": config_file["actions"]["available_actions"][
+            config_file["actions"]["actions_set"]
```
Comment (rubenlucas93) [inline rl_studio/main_rlstudio.py:21]: Se me hace un poco dificil leer qué está pasando aquí. Podría tener sentido guardar en memoria una variable actions_set = config_file["actions"]["actions_set"] y utilizarla en esta y en la siguiente asignación?

Por curiosidad. Qué diferencia hay entre available_actions y actions_set?

**Comment by rubenlucas93** [comment]
Todo guay. Puedes contestarme a los comentarios (básicamente por alinearnos en el estilo de programar y enterarme de algunas cosas que aún no he visto del f1)
La semana que viene le hecho un ojo al algoritmo que ya le tengo ganas :dancers: 

PD:
Puedes mantener los ficheros de configuración de robot_mesh, mountain_car y cartpole?
Es para facilitar que no tenga que guardármelos en local y reemplazarlos cada vez que tenga que hacer una prueba.

Gracias!

File: rl_studio/agents/__init__.py (hunk 1 above)
Comment (pjfernandecabo) [reply rl_studio/agents/__init__.py:86]: pues la que consta ahora que envié fue porque el pylint de Visual Code lo cambió. La verdad es que hace lo que quiere .  Estoy de acuerdo en tener coherencia y estoy leyendo el pep 8 de python cada vez que tengo dudas https://peps.python.org/pep-0008/, pero luego el pylint ese me lo cambia.

File: rl_studio/main_rlstudio.py (hunk 2 above)
Comment (pjfernandecabo) [reply rl_studio/main_rlstudio.py:21]: esto sirve para usar las acciones continuas. Estaba viendo que avaliable_actions contiene el arbol de las acciones discretas que definio Nacho, y las continuas que alimenta a DDPG. 
De lo que comentas de actions_set, lo pensamos y vemos. Todo ser ira aclarando segun vayamos llenando de opciones de las acciones

**Comment by pjfernandecabo** [comment]
Sobre tu cuestión de mantener los config_cartpole.yaml y demás, no los he tocado en este PR, siguen en el dir /config/
Deberían no desaparecer si hacemos el merge

**Comment by rubenlucas93** [comment]
Ya lo vi!!! Confusión mía. Gracias!!!

**Review by rubenlucas93 — APPROVED** [review]
